File: _04_MacAddress.java
import java.util.*;
import java.util.stream.*;

class _04_MacAddress {
  private static String toMacAddress() {
    var random = new Random();
    return Stream.generate(() -> random.nextInt(256))
      .limit(6)
      .map(n -> {
        String hex = Integer.toHexString(n);
        return (hex.length() == 1) ? "0" + hex : hex; 
      })
      .collect(Collectors.joining(":"));
  }
  
  public static void main(String[] args) {
    var howMany = 10;
    List<String> macAddresses = 
    Stream.iterate(0, x -> x + 1)
      .limit(howMany)
      .collect(ArrayList<String>::new, 
        (acc, n) -> acc.add(toMacAddress()),
        ArrayList::addAll);              
    System.out.println(macAddresses);
  }
}

File: 04_MacAddress.scala
import scala.util.Random

def toHex(n: Int): String = {
  val hex = n.toHexString
  if (hex.length == 1) "0" + hex else hex
}

def macAddresses(howMany: Int): List[String] = {
  val r = new Random
  Stream.continually(r.nextInt(256))
        .map(toHex)
        .take(howMany * 6)
        .grouped(6)
        .map(macAddress => macAddress.mkString(":"))
        .toList
}

println(macAddresses(10))